src/canvas/Canvas.ts

```diff
@@ -114,8 +115,13 @@ export class Canvas extends SelectableCanvas implements CanvasOptions {
 
   textEditingManager = new TextEditingManager(this);
 
-  constructor(el?: string | HTMLCanvasElement, options: TCanvasOptions = {}) {
-    super(el, options);
+  constructor(
+    arg0?: string | HTMLCanvasElement | DOMManagerType,
+    options: TCanvasOptions = {}
+  ) {
+    super(arg0, options);
+    // enable drag and drop
+    this.elements.items.top.el.setAttribute('draggable', 'true');
```

Comment:
moved from CanvasDOMManager

Reply:
not sure it is good here because my intention is to avoid polluting fabric with web specific stuff

Reply:
this are all things that needs to be discussed indeed.
Fabric is not polluted by web specific stuff. FabricJS is mainly web based, with a node backend abstracted through JSDOM. 
Correct?
So you can't avoid polluting something that is made in that way.

Reply:
True
I would argue that in the long run we can consider dropping JSDOM so I am being cautious 
Moreover, since I want Offscreen support this is considered pollution

Reply:
yes it would be good to slowly pack dom support somewhere and have fabricjs domless. But don't write your code with that in mind, when it will be done we will have to change a lot of stuff anyway.
Think of your code so is small and clear and less code than what you got in your first iteration.